blog/2020-10-08-debezium-community-stories-with-renato-mefi.adoc

```diff
@@ -0,0 +1,143 @@
+* _Dual writes_, well, actually this is not an option, because as you know, https://thorben-janssen.com/dual-writes/["Friends Don't Let Friends Do Dual Writes"]!
+* Different approaches of _distributed transactions_ like 2PC and sagas; the costs here are performance and engineering effort, now every service I have has to either become a transaction coordinator or have rollback capabilities, also the cascade effect scares us quite a bit!
+
+Well, what's left? We found that outbox was the right answer for us, but before we get there, let me get into the cost x benefit equation of our decision making!
+
+Although some of the options were quite attractive technically, for instance event sourcing, the engineering effort and growth is immense. Also, it's not the kind of thing which comes ready to use, and there's a lot of discovery to be made along the way, so what were the constraints and desires:
+
+* _Reliability_; we want at least once semantics, exactly once isn't necessary as we can uniquely identify each message/event;
+* _Eventual consistency only between services_, but not within the services themselves. Being able to interact with a service which is the source of truth of a certain model, and get an immediate answer is not just handy, but incredibly powerful;
+* _Avoiding distributed transactions_ as much as we can, it's scary and we should be scared about it too!
+* _Manageable effort_; how can we "easily" get 30+ engineers to adopt a solution for this problem? At the same time, how can you ensure the implementation guarantees among every service and team?
+
+We realized that the outbox pattern would help us meet those requirements: applications would publish events via an outbox table, which gets written to as part of business transactions in the database.
+
+As with the strangler pattern, we wanted to resort to an architecture component, something the teams could self-service. At first, we were exploring a home-grown solution which would look for the outbox tables among every service and publish the messages. The problem with this approach would be the polling databases problem, although in this case this is less harmful as we don't need to look for updates or deletes.
+
+Luckily, by that time I was closely following the work being done in Debezium and I read the blogpost about link:/blog/2019/02/19/reliable-microservices-data-exchange-with-the-outbox-pattern/[reliable data exchange between microservices using the outbox pattern], and there was my answer! Well, I mean, parts of the answer, we still needed to implement it, and that's a story for the next question!
+
+Fast forward a couple of months and we got a reliable way to exchange messages between services, with all the guarantees we wanted to have, and by applying some platform DeVops flavor to it, we also made it self-service and easy to plug in every service!
```

Comment:
"we also made it self-service and easy to plug in every service" -- would you happen to have a screenshot of that, too?